### PR TITLE
Add cloudfront url to ShareVideoDialog

### LINKED ISSUE #754: allow sharing of HLS streaming URL on specifed collections
Part of https://github.com/mitodl/mitxpro/issues/16

As an open edX course author, I'd like to be able to get the HLS streaming URL for my video, so I can paste it into a video xBlock on open edX. 

### acceptance criteria

- [ ] add a boolean field to collection model `allow_share_openedx` 
- [ ] display the boolean field in the collection admin
- [ ] if `allow_share_openedx` is true, display the full URL for the video's HLS video file when a collection owner clicks the share button in a "open edX Video URL" box, similar to the existing one for "Video URL". For example, for the URL at https://video.odl.mit.edu/videos/05abcf60f486410bbbb6cbf06463898c/ the open edX Video URL would be https://d3tsb3m56iwvoq.cloudfront.net/transcoded/05abcf60f486410bbbb6cbf06463898c/video__index.m3u8

### PR DESCRIPTION
**Pull request by annagav**
#### What are the relevant tickets?
Fix #754

#### What's this PR do?

Add cloudfront url to ShareVideoDialog

#### How should this be manually tested?
Run migrations, set allow_share_openedx on the collection to `True`, and try to share the video.

Screen Shot 2019-06-12 at 10 14 13 AM

### CODE REVIEW

**Comment by pdpinch** [comment]
@annagav as described in the issue, I want to make sure we're grabbing the HLS video file (assuming there is one). I can't tell from your code how/if we're doing that.

**Comment by annagav** [comment]
@pdpinch, I addressed your comment.

File: ui/serializers.py [hunk 1]
```diff
@@ -135,6 +137,14 @@ def validate_view_lists(self, value):
         """
         return validate_moira_lists(value)
 
+    def get_cloudfront_url(self, obj):
+        """Get cloudfront_url"""
+        video_file = obj.videofile_set.filter(encoding=EncodingNames.HLS).first()
+        if obj.collection.allow_share_openedx and video_file:
```
Comment (mbertrand) [inline ui/serializers.py]: I think this is supposed to be viewable only by the collection owner (and maybe people on the admin moira lists)?

Comment (mbertrand) [reply ui/serializers.py]: I think `SimpleVideoSerializer` also needs to be modified, if I click "Share" for a video from the collections page, the cloudfront url field doesn't appear:

Screen Shot 2019-06-17 at 4 52 07 PM

File: ui/serializers_test.py [hunk 2]
```diff
@@ -108,41 +109,75 @@ def test_collection_list_serializer():
...
         'collection_view_lists': [],
         'view_lists': [],
-        'sources': [],
+        'sources': video.sources,
         'is_private': False,
-        'is_public': public,
-        'youtube_id': (video.youtube_id if youtube and public else None)
+        'is_public': video.is_public,
+        'youtube_id': None,
+        'cloudfront_url': "",
```
Comment (mbertrand) [inline ui/serializers_test.py]: Minor nitpick, you can add these lines here:
```python
'videofile_set': serializers.VideoFileSerializer(video.videofile_set.all(), many=True).data,
'videothumbnail_set': serializers.VideoThumbnailSerializer(video.videothumbnail_set.all(), many=True).data
```

and then you don't have to set `expected['videofile_set']` or `expected['videothumbnail_set']` in the two tests that use this function.

**Comment by annagav** [comment]
@mbertrand, I addressed your comments. Thanks!

**Comment by annagav** [comment]
@mbertrand sorry, just noticed that there is new feedback :)

**Pull request merged**